app/views/shops/show.html.erb

```diff
@@ -0,0 +1,19 @@
+<% if @company.id === @shop.company_id  %>
```

Comment:
this if should not be necessary if the models are loaded correctly in the controller